src/py42/_internal/clients/alerts.py

```diff
@@ -73,19 +81,33 @@ def get_all_rules(self, sort_key=u"CreatedAt", sort_direction=u"DESC"):
             self._get_alert_rules,
             u"ruleMetadata",
             tenant_id=tenant_id,
+            groups=None,
             sort_key=sort_key,
             sort_direction=sort_direction,
         )
 
-    def get_rules_by_name(self, rule_name):
-        rule_pages = self.get_all_rules()
-        matched_rules = []
-        for rule_page in rule_pages:
-            rules = rule_page[u"ruleMetadata"]
-            for rule in rules:
-                if rule_name.lower() == rule[u"name"].lower():
-                    matched_rules.append(rule)
-        return matched_rules
+    def get_all_rules_by_name(self, rule_name, sort_key=u"CreatedAt", sort_direction=u"DESC"):
+        tenant_id = self._user_context.get_current_tenant_id()
+        return get_all_pages(
+            self._get_alert_rules,
+            u"ruleMetadata",
+            tenant_id=tenant_id,
+            groups=[json.loads(str(create_eq_filter_group(u"Name", rule_name)))],
+            sort_key=sort_key,
+            sort_direction=sort_direction,
+        )
+
+    def get_rule_by_observer_id(self, observer_id, sort_key=u"CreatedAt", sort_direction=u"DESC"):
```

Comment:
Does this not ever return more rules? Can a rule have multiple observer ids?

Reply:
nope